### DISCUSSION TITLE
MacOS drops SSH GitHub access all the time.

### DISCUSSION THREAD

**Discussion by vincej** [community/community#34774] [Other Feature Feedback, Questions, & Ideas] [answered]
Using MacOS Monterey 12.6 on a Mac Book Pro. 

Every time I do a push I get the error: git@github.com: Permission denied (publickey)
And so every time I have to enter into the terminal the below, and then it works again for a few hours, after which it it once agains fails. 

`ssh-add -l -E sha256
ssh-add ~/.ssh/id\_rsa 	
`

I can not imagine that the whole MacOs community is having this problem. 
There must be a way of fixing this.  Any ideas? 

Many thanks

**Comment by antonblr** [comment, marked as answer]
@vincej - possibly check `~/.ssh/config` for (and if it is not there - add it):
```
Host *
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_rsa
```

More details in [Adding your SSH key to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)

**Reply by vincej** [reply]
Thank you - it has been 24 hours, and they have not dropped YET!

**Comment by airtower-luna** [comment]
That doesn't sound like a bug, just like there's a default timeout of a few hours after which ssh-agent drops keys as a security measure. Setting `AddKeysToAgent` as @antonblr suggested should make it so SSH loads the key automatically (after asking for your password, if necessary).
